### DISCUSSION TITLE
METPlus Tutorial Environment on Cheyenne not working

### DISCUSSION THREAD

**Discussion by cohenjt** [dtcenter/METplus#2407] [Existing Builds] [answered]
Hi! I am trying to follow the tutorial here: https://dtcenter.org/metplus-practical-session-guide-version-5-0/session-1-metplus-setupgrid-grid/metplus-setup/metplus-initial-setup/setting-tutorial-environment and am running my code on Casper. I followed these directions, but when I ran `source METplus-5.0.0_TutorialSetup.sh`, a number of modules could not load as it seemed the version numbers were out of date:

```Lmod has detected the following error:  The following module(s) are unknown: "ncarenv/1.3" "netcdf/4.8.0"```

I think this is just caused by the outdated version numbers in the config file, but I'm not sure how to fix this to get the environment running correctly. Thanks for your help!

**Comment by georgemccabe** [comment]
Hi @cohenjt,

The METplus 5.0 tutorial instructions were tested using Cheyenne, but they were not tested on Casper. If you are able to use Cheyenne instead of Casper, I would suggest doing that.

There are instructions to set up METplus on Casper on the [METplus-5.0 Existing Builds](https://dtcenter.org/community-code/metplus/metplus-5-0-existing-builds) page. However it appears that the modules on Casper have changed since the Casper-specific METplus module files were created, because the metplus/5.0.1 and met/11.0.1 module scripts were broken. I worked with another METplus developer just now to fix these module scripts so they do not produce errors (thanks @jprestop !), but we have not tested that everything in the tutorial will work as expected. Also, other METplus components like METcalcpy and METplotpy were not configured for casper, so tutorial instructions that use those components will not work.

If you would like to try using the tutorial on Casper, you will need to make a few modifications to the tutorial setup script after you copy it into your directory, described below. Again, please keep in mind that the tutorial has not been tested to run this way and some portions of the tutorial will likely not work as expected.

Change the module use command for METplus to (previously did not have casper directory):
`module use /glade/p/ral/jntp/MET/METplus/casper/modulefiles` 

and remove the METcalcpy, METplotpy, and METdataIO module loads:
```
module use /glade/p/ral/jntp/MET/METcalcpy/modulefiles
module load metcalcpy/2.0.2
module use /glade/p/ral/jntp/MET/METplotpy/modulefiles
module load metplotpy/2.0.1
module use /glade/p/ral/jntp/MET/METdataio/modulefiles
module load metdataio/2.0.2
```

**Reply by ahijevyc** [reply]
Just in case someone else familiar with the old met versions has the same problem after running `module use /glade/p/ral/jntp/MET/METplus/casper/modulefiles`
you might need to reset your modules `module reset` to remove any wrong module locations (i.e. /glade/p/ral/jntp/MET/MET_releases/modulefiles). I still had the old one in my path from the previous attempts. And you have to load `metplus` not `met`.

**Reply by jprestop** [reply, marked as answer]
Hi @ahijevyc.  I wanted to let you know that we have updated our installations on Derecho and Caper and we are now using modulefiles in new locations.  You can find that information on the [Existing Builds](https://dtcenter.org/community-code/metplus/metplus-5-1-existing-builds) page for the coordinated 5.1 release under "NCAR machines".  Please let us know if you have any questions or problems.  If you do not have any further questions, we'd love to have you select "Mark as Answer" for the answer you feel best resolved the problem and then I'll lock this Discussion to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we’ll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that’ll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion.

**Reply by cohenjt** [reply]
Hi @jprestop. Thank you for your comment. I have been successfully running MODE with python embedding. I used the new paths from the existing builds page to change the bash file I use to setup MODE to the following. I think I did something incorrectly because MODE is not running anymore. Please let me know if the mistake is easy to identify, otherwise I can upload my config files and the resulting log file. Thank you!

```
# METplus v5.0.0 Tutorial Setup Script on Cheyenne (NCAR) using bash

# module use /glade/p/ral/jntp/MET/METplus/casper/modulefiles
module use /glade/work/dtcrt/METplus/casper/components/METplus/installations/modulefiles
module load metplus/5.1.0

module use /glade/work/dtcrt/METplus/casper/components/MET/installations/modulefiles
module load met/11.1.0

# Directory to store tutorial files - configurations, output, notes, etc.
export METPLUS_TUTORIAL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"

# Path to the METplus installation location
# was /glade/p/ral/jntp/MET/METplus/METplus-5.0.1
export METPLUS_BUILD_BASE=/glade/work/dtcrt/METplus/casper/components/METplus/installations/METplus-5.1.0

# Path to the MET installation location
# was /glade/p/ral/jntp/MET/MET_releases/11.0.1
export MET_BUILD_BASE=/glade/work/dtcrt/METplus/casper/components/MET/installations/11.1.0

# Path to input data required for running use cases
# was /glade/p/ral/jntp/MET/METplus/METplus-5.0_sample_data
export METPLUS_DATA=/glade/work/dtcrt/METplus/data/components/METplus/METplus-5.1_sample_data
```

**Reply by jprestop** [reply]
Hi @cohenjt.  All of those module paths look correct to me.  I loaded met/11.1.0 and metplus/5.1.0 and both seemed to have worked as expected.  Please pass along your config files and log file.  Hopefully, the log file will contain an error that we can take a look at and try to resolve.

**Reply by cohenjt** [reply]
Hi @jprestop. I just uploaded all my files to the server under cohen_data. Please let me know if there's anything else I can send over. The log file is ~15MB and seems to contain a lot of error messages that I am having trouble understanding. Thank you for your help!

**Reply by jprestop** [reply]
@cohenjt Thank you so much for uploading your log file.  It was very helpful.  I see the error:
```
ImportError: /glade/work/dtcrt/METplus/casper/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/netCDF4/../../../libnetcdf.so.19: undefined symbol: H5Pset_fapl_ros3
```
which I have seen before.  We have addressed this problem in our MET-12.0.0-beta3 release.  I'm going to talk with my team about the best way to proceed.  I will follow up once I have more information.  In the meantime, please feel free to switch back to what you were using previously, but at some point soon, I know that access to /glade/p will be going away.

**Reply by cohenjt** [reply]
Hi @jprestop. Thank you again for your help! I returned to using the old file paths to /glade/p/, and the log files show that MODE was able to start correctly, but eventually reached the same error as above. I uploaded one of the new log files if you want to take a look.

**Reply by jprestop** [reply]
@cohenjt In your comments [here](https://github.com/dtcenter/METplus/discussions/2407#discussioncomment-8389215) it looks like you were using MET-11.0.1, but in the current log file it looks like you're still using MET-11.1.0 in the new location:

> 02/07 20:28:50.955Z metplus INFO: COMMAND: /glade/work/dtcrt/METplus/casper/components/MET/installations/11.1.0/bin/mode

If you only run:

> module use /glade/p/ral/jntp/MET/METplus/casper/modulefiles
> module load metplus/5.0.1

It should load met/11.0.1 for you, so you should be all set.  Please give that a try and see if it's working for you.

**Reply by cohenjt** [reply]
@jprestop Turns out I forgot to switch the path in the MODEConfig_dir_final.conf file. It's updated now and working again, thank you!

**Reply by jprestop** [reply]
@cohenjt Great!  Thanks for letting us know. I'm glad it's working for you again.

**Reply by jprestop** [reply]
@cohenjt I implemented a change in the compilation for MET-11.1.0.  If it's not too much trouble, could you please try switching back to the new versions and letting me know if it works as expected?  If it does not, could you please pass along your log file again?

**Reply by cohenjt** [reply]
@jprestop It seems to be working now, thank you! I will mark your first comment as the answer to this discussion.

**Reply by cohenjt** [reply]
@jprestop the only other thing to note, which doesn't seem to be causing any issues, is a warning that repeatedly shows up in the log file that I don't recall seeing before:
```
/glade/work/dtcrt/METplus/casper/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/numpy/core/getlimits.py:518: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/glade/work/dtcrt/METplus/casper/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/numpy/core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
```

Just wanted to document this in case you recognize it or it causes problems for someone else. Thanks again!

**Reply by jprestop** [reply]
Thank you for letting us know @cohenjt.  We are not familiar with this warning.  Hopefully, this won't cause you any trouble.  We are handing the initial error differently for our 12.0.0-beta3 release, which I should be installing soon (after the METplus beta3 version is released).  If you'd like me to let you know when the beta3 installations are complete so that you can try those out, please let me know and I'll follow up.  I'll also be updating the [Existing Builds page for our coordinated 6.0 release](https://dtcenter.org/community-code/metplus/metplus-6-0-existing-builds).  Otherwise, I'll lock this Discussion now to prevent future posts on this thread.
